Correct pip chdir without virtualenv

tonal opened the pull request:
see #37912

##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
pip modile broken when set option `chdir` without `virtualenv`
<!---
If you are fixing an existing issue, please include "Fixes #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Bugfix Pull Request

##### COMPONENT NAME
<!--- Name of the module, plugin, module or task -->
lib/ansible/modules/packaging/language/pip.py

##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes below -->
```
ansible 2.5.0
  config file = /home/user/projects/remains/deploy/ansible.cfg
  configured module search path = [u'/home/user/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.12 (default, Dec  4 2017, 14:50:18) [GCC 5.4.0 20160609]
```

tonal committed:
Correct pip chdir without virtualenv

see #37912

Akasurde commented on lib/ansible/modules/packaging/language/pip.py:
```diff
@@ -384,7 +384,7 @@ def main():
     venv_created = False
     if chdir:
-        env = os.path.join(chdir, env)
+        env = os.path.join(chdir, env or '')
```
Do you think we should add additional check for `env` variable so that it doesn't silently takes blank value ?

robinro replied:
Imo having `type=path` is enough validation for `virtualenv` or what kind of blank values do you mean?

mdellweg replied:
I am not sure, this solves the problem as intended, because it might accidentially set env to some path (chdir) if it was supposed to be None (aka do not use / create a virtual env).

robinro replied:
As a reference, @mdellweg suggests https://github.com/ansible/ansible/pull/39595/files
I like the solution by @mdellweg more. @tonal what's your opinion?

tonal replied:
I think @mdellweg is right.
How to about both set `chdir` and `executable`?
It seems that in this case, too, you need to joined the paths:
```python
    executable = module.params['executable']
    if chdir:
        if env:
            env = os.path.join(chdir, env)
        if executable:
            executable = os.path.join(chdir, executable)
### and after
    pip = _get_pip(module, env, executable)
```

robinro reviewed (approved):
Thanks @tonal.
The change looks good to me. Maybe you can add a testcase to `test/integration/targets/pip/tasks/pip.yml`? The test coverage for `chdir` is pretty bad.

robinro commented:
Some tests complain about missing `git`, no idea atm how that worked so far, but this needs to be fixed before the merge.

tonal committed:
Test case for #37912

Akasurde commented on test/integration/targets/pip/tasks/pip.yml:
```diff
@@ -202,3 +202,16 @@
+  pip:
+    name: pip
+    virtualenv: "{{ output_dir }}/pipenv"
+  
```
Can you please add assertion like rest of integration tests here as well ?

tonal committed:
assert for test

tonal commented:
@robinro git remove from requirements. See #35197 and #35885

tonal committed:
Add install git

tonal committed:
Do not install git on MacOSX

julienvey committed:
ec2_instance: add a retry to run_instance to help with ec2 consistency

abadger committed:
Categorize FILE_COMMON_ARGS

Some file_common_args are not common to all file modules.  Divide the
args into categories for a future refator of this into a separate
file_argspec.

abadger committed:
Make the copy module pass correct parameters to file module

abadger committed:
Initial cleanup of file module

* Remove use of six.b as Python-2.6+ have byte literals.
* Make AnsibleModule a global object so we'll have access to it in all
  the functions we're going to break this up into.
* Rework the parameters so things that are in file_common_args are used
  from file_common_args or the reason for deviation is documented.
* Remove validate as a parameter: this should be taken care of by
  removing it from params before the copy and template action plugin
  invoke file.
* Rename diff_peek to _diff_peek as it is an internal parameter.
* add module_name execute_module call to assemble so that it is more greppable

abadger committed:
Document follow changes

Add the file, blockinfile, and replace changes to the follow parameter
to the porting guide

abadger committed:
Fix encoding issue when parsing plugin examples

This caused a non-fatal traceback when building, for example, the module
documentation for the debug module.

robinro commented:
@tonal 
Please rebase to integrate #39460

abadger committed:
Fix for file module with recursive permission setting and broken symlinks

There was a traceback when setting permissions on a directory tree when
there were broken symlinks inside of the tree and follow=true.  chmod -R
ignores broken symlinks inside of the tree so we've fixed the file
module to do the same.

Fixes #39456

robinro commented:
@tonal Can you address the two issues raised in review?
I'd like to see this merged soon. Feel free to contact me or #ansible-devel on IRC if you need assistance.

nerzhul committed:
New module: AWS storage gateway facts (#39491)

* Add a module to get storage gateway facts

* Review fixes

* Last review fixes

* Add filtering gathering & some fixes

* doc fix

* API error handling

* Remove ec2_argument_spec import

Use imported BotoCoreError and ClientError rather than botocore.exceptions

ganeshrn committed:
Fix eos_config integration test failure (#39662)

Enable become for eos_config check_mode test cases

ganeshrn committed:
Fix eos_vlan associated interface check (#39661)

* Fix eos_vlan associated interface check

Fix eos_vlan associated interface check by comparing
the interface in want and have without converting the
interface name to lower

* Update eos_vlan docs

s-hertel committed:
ec2_ami: Properly delete snapshots (#39606)

cahlchang committed:
[aws] Add aws_iam_role check mode support (#39002)

* Check mode when adding

* Check mode when deleting

* Add check mode

willthames committed:
[aws] Improve integration test suite for cloudfront (#37813)

Improve naming of one of the cloudfront tasks

Change test_identifier back to resource_prefix now it's always
lower case.

More tests around using distribution_id and default_root_object

roadmapper committed:
Fix ec2_metadata_facts documentation (#37546)

- ec2_metadata_facts does not extend `url`
- Remove url_argument_spec for ec2_metadata_facts

felixfontein committed:
Route53 check mode (#37273)

* Adding check mode for route53.

* flake8

willthames committed:
Change aws_acm_facts name parameter to be domain_name (#36986)

Match AWS's naming for domain name, but provide backward
compatible alias.

ooocamel committed:
[doc] fix a typo in ec2_eni docs (#36571)

willthames committed:
Fail with nice error message if elb target_type=ip not supported (#38313)

* Add helpful failure message if target_type=ip is not supported

Create test case for target_type=ip not supported

* Update elb_target_group module to latest standards

Use AnsibleAWSModule
Improve exception handling
Improve connection handling

s-hertel committed:
[cloud] aws_direct_connect_connection: improve return docs (#37677)

Fixes #29381

Fix uses of AWSRetry

Fix exception handling

Update module to use AnsibleAWSModule

gundalow committed:
Default ssh port (#39678)

ruizink committed:
Add new OpenStack module for managing cinder volume snapshots (#19122)

* Add os_volume_snapshot Ansible Module

* Add missing RETURN documentation

* Fix YAML syntax

* Add ANSIBLE_METADATA

Signed-off-by: Mário Santos <mario.rf.santos@gmail.com>

* Bump the ansible version

Signed-off-by: Mário Santos <mario.rf.santos@gmail.com>

* Add support for check_mode

Signed-off-by: Mário Santos <mario.rf.santos@gmail.com>

* Fix sanity tests

Signed-off-by: Mário Santos <mario.rf.santos@gmail.com>

dharmabumstead committed:
Adds documentation structure and stubs for new VMWare guide (#39246)

* Adding initial skeleton of VMWare docs

* Updated product name capitaliztion; ready for initial PR.

* Add few files and updated some links

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

* Fixed duplicate anchor tag

seuf committed:
Don't split line when no version available for plugin (#37907)

* Don't split line when no version available for plugin

* Fixed author github account

maxamillion committed:
fix sanity checks to make shippable happy (#39685)

Signed-off-by: Adam Miller <admiller@redhat.com>

Commit:
fix nxos_snmp_traps issues (#39444)

* fix snmp_traps code

* add IT cases

* fix shippable

* fix shippable without ignore

Commit:
Fix for nxos_snmp_host issues (#39642)

* fix snmp_host issues

* source files

* fix shippable

* remove defaults to match arg spec

gundalow committed:
vyos_interface require multiple network nodes to run (#39670)

* vyos_interface require multiple network nodes to run

We don't have the ability to run these currently, so disable them.
The original logic was also incorrect, the tests don't pass on lab, DCI
nor single instance nodepool, so disable

https://github.com/ansible/ansible/issues/39667 tracks getting these
enabled again

* eth0 -> Gi0/0

* Correctly detect if we should run lldp

* Correctly detect if we should run lldp

s-hertel committed:
[ec2_vpc_vgw] [ec2_vpc_vpn] stabilize modules for PR 35983 (#38666)

* Stabilize ec2_vpc_vgw and ec2_vpc_vpn so tests for ec2_vpc_vpn_facts in PR 35983 can be run in CI

* Add updated placebo recordings

* ensure find_vgw uses the virtual gateway id if available

Add AWSRetry.jittered_backoff to attach_vpn_gateway to deal with errors when attaching a new VPC directly after detaching

Add integrations tests for ec2_vpc_vgw

* Sort VPN Gateways by ID

gundalow committed:
Use ansible-test, not nxos.yaml (#39692)

mattclay committed:
Enable nuage_vspk tests for Python 2.7.

sivel committed:
Restore previous test play, break new tests into a new play (#39688)

ZuZuD committed:
ec2_vpc_route_table: Update matching_count parsing on find_subnets fu… (#38707)

* ec2_vpc_route_table: Update matching_count parsing on find_subnets function and tests

* ec2_vpc_route_table: Update matching_count parsing on find_subnets function

kbreit committed:
Meraki organization module (#38773)

* Initial commit

Query an organization within Meraki. No support is in place for managing
or creating yet

* Change output_level method and make the state parameter required.

* Implemented listing all organizations

- Updated documentation
- Parse results and return all organizations
- Parse results and return specified organization

* Framework for creating an organization

- Documentation example for organization creation
- Framework exists for creating organizations, pending PR 36809
- Created functions for HTTP calls
- Renamed from dashboard.meraki.com to api.meraki.com
- Added required_if for state

* Remove absent state

- Meraki API does not support deleting an organization so absent is removed
- Updated documentation to call it state instead of status

* Small change to documentation

* Support all parameters associated to organization

- Added all parameters needed for all organization actions.
- None of the added ones work at this time.
- Added documentation for clone.

* Integration test for meraki_organization module

* Rename module to meraki for porting to module utility

* Meraki documentation fragment

- Created initial documentation fragment for Meraki modules

* Add meraki module utility to branch. Formerly was on a separate branch.

* CRU support for Meraki organization module

* CRU is supported for Meraki organizations
* There is no DELETE function for organizations in the API
* This code is very messy and needs cleanup
* Create and Update actions don't show status as updated, must fix

* Added Meraki module utility to module utility documentation list

* Added support for organization cloning
* Renamed use_ssl to use_https
* Removed define_method()
* Removed is_org()
* Added is_org_valid() which does all org sanity checks

* Fixes for ansibot
- Changed default of use_proxy from true to false
- Removed some commented out code
- Updated documentation

* Changes for ansibot
- Removed requirement for state parameter. I may readd this.
- Updated formatting

diff --git a/lib/ansible/module_utils/network/meraki/meraki.py b/lib/ansible/module_utils/network/meraki/meraki.py
index 3acd3d1038..395ac7c4b4 100644
--- a/lib/ansible/module_utils/network/meraki/meraki.py
+++ b/lib/ansible/module_utils/network/meraki/meraki.py
@@ -42,7 +42,7 @@ def meraki_argument_spec():
     return dict(auth_key=dict(type='str', no_log=True, fallback=(env_fallback, ['MERAKI_KEY'])),
                 host=dict(type='str', default='api.meraki.com'),
                 name=dict(type='str'),
-                state=dict(type='str', choices=['present', 'absent', 'query'], required=True),
+                state=dict(type='str', choices=['present', 'absent', 'query']),
                 use_proxy=dict(type='bool', default=False),
                 use_https=dict(type='bool', default=True),
                 validate_certs=dict(type='bool', default=True),
diff --git a/lib/ansible/modules/network/meraki/meraki_organization.py b/lib/ansible/modules/network/meraki/meraki_organization.py
index 923d969366..3789be91d6 100644
--- a/lib/ansible/modules/network/meraki/meraki_organization.py
+++ b/lib/ansible/modules/network/meraki/meraki_organization.py
@@ -20,11 +20,9 @@ short_description: Manage organizations in the Meraki cloud
 version_added: "2.6"
 description:
 - Allows for creation, management, and visibility into organizations within Meraki
-
 notes:
 - More information about the Meraki API can be found at U(https://dashboard.meraki.com/api_docs).
 - Some of the options are likely only used for developers within Meraki
-
 options:
     name:
         description:
@@ -32,21 +30,18 @@ options:
         - If C(clone) is specified, C(name) is the name of the new organization.
     state:
         description:
-        - Create or query organizations
-        choices: ['query', 'present']
+        - Create or modify an organization
+        choices: ['present', 'query']
     clone:
         description:
         - Organization to clone to a new organization.
-        type: string
     org_name:
         description:
         - Name of organization.
         - Used when C(name) should refer to another object.
-        type: string
     org_id:
         description:
         - ID of organization
-
 author:
     - Kevin Breit (@kbreit)
 extends_documentation_fragment: meraki
@@ -86,7 +81,6 @@ RETURN = '''
 response:
     description: Data returned from Meraki dashboard.
     type: dict
-    state: query
     returned: info
 '''

@@ -103,6 +97,7 @@ def main():
     argument_spec = meraki_argument_spec()
     argument_spec.update(clone=dict(type='str'),
+                         state=dict(type='str', choices=['present', 'query']),
                          )

@@ -125,11 +120,9 @@ def main():

     meraki.function = 'organizations'
     meraki.params['follow_redirects'] = 'all'
-    meraki.required_if=[
-                           ['state', 'present', ['name']],
-                           ['clone', ['name']],
-                           # ['vpn_PublicIP', ['name']],
-                       ]
+    meraki.required_if = [['state', 'present', ['name']],
+                          ['clone', ['name']],
+                          ]

     create_urls = {'organizations': '/organizations',
                    }
@@ -162,23 +155,16 @@ def main():

-
-    # method = None
-    # org_id = None
-
-
-    # meraki.fail_json(msg=meraki.is_org_valid(meraki.get_orgs(), org_name='AnsibleTestOrg'))
-
     if meraki.params['state'] == 'query':
-      if meraki.params['name'] is None:  # Query all organizations, no matter what
-        orgs = meraki.get_orgs()
-        meraki.result['organization'] = orgs
-      elif meraki.params['name'] is not None:  # Query by organization name
-        module.warn('All matching organizations will be returned, even if there are duplicate named organizations')
-        orgs = meraki.get_orgs()
-        for o in orgs:
-          if o['name'] == meraki.params['name']:
-            meraki.result['organization'] = o
+        if meraki.params['name'] is None:  # Query all organizations, no matter what
+            orgs = meraki.get_orgs()
+            meraki.result['organization'] = orgs
+        elif meraki.params['name'] is not None:  # Query by organization name
+            module.warn('All matching organizations will be returned, even if there are duplicate named organizations')
+            orgs = meraki.get_orgs()
+            for o in orgs:
+                if o['name'] == meraki.params['name']:
+                    meraki.result['organization'] = o
     elif meraki.params['state'] == 'present':
         if meraki.params['clone'] is not None:  # Cloning
             payload = {'name': meraki.params['name']}
@@ -193,7 +179,10 @@ def main():
             payload = {'name': meraki.params['name'],
                        'id': meraki.params['org_id'],
                        }
-            meraki.result['response'] = json.loads(meraki.request(meraki.construct_path('update', org_id=meraki.params['org_id']), payload=json.dumps(payload), method='PUT'))
+            meraki.result['response'] = json.loads(meraki.request(meraki.construct_path('update',
+                                                                                        org_id=meraki.params['org_id']),
+                                                                  payload=json.dumps(payload),
+                                                                  method='PUT'))

diff --git a/lib/ansible/utils/module_docs_fragments/meraki.py b/lib/ansible/utils/module_docs_fragments/meraki.py
index e268d02e68..3569d83b99 100644
--- a/lib/ansible/utils/module_docs_fragments/meraki.py
+++ b/lib/ansible/utils/module_docs_fragments/meraki.py
@@ -35,6 +35,7 @@ options:
         description:
         - Set amount of debug output during module execution
         choices: ['normal', 'debug']
+        default: 'normal'
     timeout:
         description:
         - Time to timeout for HTTP requests.
diff --git a/test/integration/targets/meraki_organization/aliases b/test/integration/targets/meraki_organization/aliases
new file mode 100644
index 0000000000..ad7ccf7ada
--- /dev/null
+++ b/test/integration/targets/meraki_organization/aliases
@@ -0,0 +1 @@
+unsupported

* Formatting fix

* Minor updates due to testing
- Made state required again
- Improved formatting for happier PEP8
- request() now sets instance method

* Fix reporting of the result

* Enhance idempotency checks
- Remove merging functionality as the proposed should be used
- Do check and reverse check to look for differences

* Rewrote and added additional integration tests. This isn't done.

* Updated is_update_required method:
- Original and proposed data is passed to method
- Added ignored_keys list so it can be skipped if needed

* Changes per comments from dag
- Optionally assign function on class instantiation
- URLs now have {} for substitution method
- Move auth_key check to module utility
- Remove is_new and get_existing
- Minor changes to documentation

* Enhancements for future modules and organization
- Rewrote construct_path method for simplicity
- Increased support for network functionality to be committed

* Changes based on Dag feedback and to debug problems

* Minor fixes for validitation testing

* Small changes for dag and Ansibot
- Changed how auth_key is processed
- Removed some commented lines
- Updated documentation fragment, but that may get reverted

* Remove blank line and comment

* Improvements for testing and code simplification
- Added network integration tests
- Modified error handling in request()
	- More testing to come on this
- Rewrote construct_path again. Very simple now.

* Remove trailing whitespace

* Small changes based on dag's response

* Removed certain sections from exit_json and fail_json as they're old

bcoca committed:
rebase base playbook base (#39533)

* rebase base playbook base

fixes issues with loop control allowing generic attributes it shouldn't

jborean93 committed:
win_get_url: Fixed a few issues with using FTP and added tests (#39646)

* win_get_url: Fixed a few issues with using FTP and added tests

* Fixed typo in docs

abadger committed:
Reorganize and expand the file tests

* Pull the tests for state=link into their own file
* Pull tests for what happens when dest is a directory out
* Expand both of the above sets of tests

abadger committed:
Fix for file module with symlinks to nonexistent target (#39635)

* Fix for file module with symlinks to nonexistent target

When creating a symlink to a nonexistent target, creating the symlink
would work but subsequent runs of the task would fail because it was
trying to operate on the target instead of the symlink.

Fixes #39558

pilou- committed:
vdirect modules: fix 'import' sanity test (#39706)

* vdirect modules: fix 'import' sanity test

* Remove passing file from import skip list.

* vdirect modules: fix validate-modules warnings

- Arguments with a default should not be marked as required
- add choices in doc

* vdirect_runnable: use formatting function

yuwzho committed:
Fixes #37731 Create Internal Loadbalancers with azure_rm_loadbalancer, support SKU (#38077)

* create internal loadbalancer

* fix test

* remove duplicate test

* clean up

* fix doc

* lint

* add sku support

* update version

* change to the version the same as CLI

* add pip support sku

* fix lint

* fix test

* Update main.yml

* add changelog entry

trishnaguha committed:
Handle nxos_feature issue where json isn't supported (#39150)

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

machacekondra committed:
ovirt: Return ID for permissions facts (#39721)

JulienPalard committed:
FIX: Removing superfluous bloc wrongly introduced in 2b7799f817e8. (#39723)

webknjaz committed:
Integrate bug/docs report and FR issue templates

ganeshrn committed:
Fix ios integration test failures in CI (#39729)

Fix ios integration test failures in CI. Since the packet transfer and receive rate
on the interface is not determined to remove the tx_rate and rx_rate test conditions
to prevent intermittent failure.

sivel committed:
Fix E126, E123 pep8 errors, and ansible-format-automatic-specification pylint error (#39740)

trishnaguha committed:
use show run instead of section pipeline ios_l2_interface (#39658)

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

webknjaz committed:
Add emojii to issue template titles

According to https://gitmoji.carloscuesta.me/

sivel committed:
Allow using action/local_action on includes and imports (#37260)

* Prevent using action/local_action on includes and imports. Fixes #28822

* Use ModuleArgsParser to determine action instead of disallowing action/local_action with import/include

* Add to_native

* switch back to block in task_ds, use ModuleArgsParse otherwise

* var should be task_ds

* Add test validating action+include_tasks

machacekondra committed:
ovirt: Don't return ID as list in fetch_nested (#39736)

* ovirt: Don't return ID as list in fetch_nested

* ovirt_permissions_facts: Fix pylint

sivel committed:
Add changelog entries for 2.6 include fixes (#39750)

mkrizek committed:
import_role docs: loop is not supported (#39725)

rambleraptor committed:
New module: GCP Compute Routes (#38619)

rambleraptor committed:
New module: GCP Compute Networks (#38618)

rambleraptor committed:
New module: GCP Compute Subnetworks (#38621)

rambleraptor committed:
New module: GCP Pubsub Subscriptions (#38626)

rambleraptor committed:
New module: GCP Pubsub Topics (#38627)

rambleraptor committed:
Bug fixes for GCP Compute Address (#38630)

rambleraptor committed:
New module: GCP Container Node Pools (#37325)

mgu committed:
Fix gathering facts in run_once play (#39453)

* Fix gathering facts in run_once play

Fixes https://github.com/ansible/ansible/issues/39312

* Check that run_once doesn't prevent fact gathering

wimnat committed:
[aws] Create classes for Application Load Balancer (#33769)

* Create classes for Application Load Balancer
* Add unsupported CI alias
* Add AWSRetry
* Add integration tests using the ALB

jborean93 committed:
base64 filter: Added ability to specify encoding (#39714)

* base64 filter: Added ability to specify encoding

* Added unicode chars for further testing

* Removed errors to keep previous behaviour in place

* Removed surrogate pairs due to issues loading YAML in CI

sivel committed:
Make sure that the implicit async_status gets the env from a task with async. Fixes #39699 (#39764)

resmo committed:
influxdb_query: fix use of common return results (#39626)

tmshn committed:
timezone module: fixed timezone identification logic in *BSD (#36715)

hyperized committed:
rabbitmq_bindings rewrite (#35651)

* Attempt to get rid of globals

* Relocate imports and make formatting non-automatic

* spacing!

danieljaouen committed:
Homebrew: Update _uninstall_current_package to use the --force option (#33360)

seuf committed:
Added alexanderzobnin-zabbix-datasource datasource type (#36948)

mattclay committed:
Temporarily disable the nuage_vspk test.

This test interferes with the uri test on centos6.

Akasurde committed:
VMware: check SwitchUUID value before assigning (#38369)

This fix adds a check for switchuuid value of distributed virtual switch.
When there is no association between hostsystem and distributed virtual portgroup,
both specified by user, then module does not find DVSwitch.
This patch tries to mitigate that problem.

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
VMware: Allow user to set customization specification (#38530)

This fix adds new argument parameter which allows user to
set customization specification which is already created with required
values like Windows Product Key and Networking details etc.

Fixes: #38404

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

ganeshrn committed:
Fix nxos terminal plugin regex (#39659)

Fix nxos terminal plugin regex to match
command prompt without newline character as prefix

sac committed:
lvg: Allow unit to be specified for pesize (#39666)

Akasurde committed:
VMware: new module: vmware_vswitch_facts (#37176)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
VMware: new module : vmware_local_user_facts (#37167)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
VMware: Refactor vmware_host_facts (#39273)

* Updated documentation
* Error handling in find_obj
* Updated return facts example

Fixes: #39265

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
VMware: new module : vmware_resource_pool_facts (#38674)

This module adds capability to provide facts about all resource
pools in given environment.

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
VMware: report failure if no snapshot exists on VM (#38651)

This fix add correct reporting of failure if VM does not contain
any snapshots for following operations - rename, remove and revert.

Fixes: #37906

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
VMware: New module: vmware_portgroup_facts (#36740)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

frntn committed:
vmware_guest: cast vlan to int or str when relevant (#39726)

`network['vlan']` should be a VLAN ID
Integers passed around using jinja variable references are
converted to strings (see # 9362)
The # 32738 PR should allow using 'NativeType' in ansible
Explicitly converting to integer will make the module works
as expected with or without the NativeType support

`network['vlan']` can also be a VLAN NAME (fallback)
Explicitly converting to string will make the module works
as expected with or without the NativeType support

Akasurde committed:
VMware: New module: vmware_tag (#37261)

This module is based on vSphere REST API. This module allows
user to manage various tags and their association with
categories. This fix also adds vCenter REST client library which can
be re-used for other REST based modules.

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

rambleraptor committed:
Adding support for GCP Compute Health Checks (#37438)

eliba committed:
Update call method for parse_cli_textfsm (#38437)

+label: docsite_pr

sivel committed:
Use updated task, instead of original, non-merged included_file._task (#39762)

* Used updated task, instead of original, non-merged included_file._task. Fixes #39637

* Add changelog entry

ErwanAliasr1 committed:
facts: Detecting NVME partitions under Linux (#39730)

In the current state of the code, the nvme partitions are returned as empty as in :
        "ansible_devices": {
            "nvme0n1": {
                "model": "SAMSUNG MZVLW256HEHP-000L7",
                "partitions": {},

The parsing of the /sys/block/<diskname> try to find a disk named like :
    <diskname><x> as in sda1 for sda

But in the nvme context, the partition of nvme0n1 is named nvme0n1p1.
This add a possible 'p' between the diskname and the partname.

This patch simply add the option of having a 'p' between the diskname
and the partname.

The patch works on my host :
                "model": "INTEL SSDPEDMD400G4",
                "partitions": {
                    "nvme0n1p1": {
                         ...
                        "size": "93.13 GB",
                    }

Fixes #38742
Signed-off-by: Erwan Velu <erwan@redhat.com>

Akasurde committed:
VMware: Allow user to add host without cluster (#39490)

This fix allows user to add ESXi host system under folder
without requiring to specify cluster name.

partially fixes: #38300

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

sivel committed:
Allow subspec defaults to be processed when the parent argument is not supplied (#38967)

* Allow subspec defaults to be processed when the parent argument is not supplied

* Allow this to be configurable via apply_defaults on the parent

* Document attributes of arguments in argument_spec

* Switch manageiq_connection to use apply_defaults

* add choices to api_version in argument_spec

sivel committed:
Pass obj=data to AnsibleParserError for context in IncludeRole. Fixes #31374 (#39805)

mattclay committed:
Restore nuage_vspk integration tests.

abadger committed:
Use a more exact regex (#39811)

The pattern we're matching can have zero or one p.  Be more careful to
match exactly that.

Slight revision of #39730

crossan007 committed:
Broken link to MSDN (#39814)

Fix broken link to MSDN site for FileSystemRights enumeration

+label: docsite_pr

sivel committed:
Add IncludeRole.get_name to provide better name for include_role. Fixes #36343 (#39807)

agaffney committed:
Update vars for loop_control on each loop iteration (fixes #38899) (#39818)

abadger committed:
Initial refactor of the file module

* Separate the logic for each state into separate functions
* Start the process of separating out initialization (pre-processing of
  parameters that cannot be done via arg spec) from the logic to
  implement each state.
* Start the process of raising exceptions for errors and returning
  result values from each state implementing function   Goal is for all
  fail_json's to be consolidated into exception handlers at the toplevel
  and for there to be only one exit_json() at the toplevel.

abadger committed:
Port away from fail_json()

Use an exception to return failures rather than fail_json().  This way
we can easily catch the failures if the calling code decides it can deal
with it.  This has the side effect of making it easier to unittest this
code as we can catch the expected exceptions instead of having to catch
the interpreter exiting and then parse stdout for the expected data.

abadger committed:
Switch from calling exit_json() to returning results

Well organized programs should only have a few successful exit points.

This commit moves all of the successful exit points for the file module
into the main() function.  Other functions return their results to the
main function which can then choose whether there is more procesing to
do before exit or not.

Commit:
allow become methods for Jail connection plugin (#31746)

* allow become methods

- i personally tested 'su', 'sudo' and 'doas'

* forgot to import constants m(

Commit:
fix 'doas' become_method support, previously committed patch not submitted to devel branch (#37511)

* fix become_method 'doas' support by properly specifying becomecmd

a repatch of https://github.com/ansible/ansible/pull/13451/ which was never committed to 'devel' branch.

* fix play_context test for become_method doas to match new becomecmd

Commit:
Add ARA Records Ansible to community tools (#39836)

ARA is an Ansible callback that records playbook run data in a local
or remote database. This data is exposed through a CLI client as well
as a reporting web application.

abn committed:
azure: fix creation of TXT dns record sets (#38368)

As per `azure.mgmt.dns.models.txt_record.TxtRecord`, expected value for a
record is of type `[str]`. Fix TXT argspec to specify type as `list`
instead of `str`.

Fixes #37581

Reference: https://docs.microsoft.com/en-us/python/api/azure.mgmt.dns.models.txtrecord?view=azure-python

ganeshrn committed:
Fetch ios default config is defaults is enabled (#39741)

If default option is eanbled fetch the current
running configuration by adding `all` or `full`
flag

gdpak committed:
Handling of configurations blocks with end-* at the end of the block (#39673)

* handle end-policy issue

* revert changes in iosxr cliconf

* fix trailing parents not included in difference

* Moving fix to platform specific fix

* pep 8 issues

trishnaguha committed:
nxos_linkagg normalize interface (#39591)

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

SoundGoof committed:
hashi_vault: fix for : in secret name (#39820)

Commit:
fix nxos_snmp_user issues (#39760)

* fix nxos_snmp_user issues

* shipppable fix

cocaccola committed:
docs: fix 'ipmi' typo (#39823)

felixfontein committed:
Renaming letsencrypt -> acme_certificate (#39816)

* Renaming letsencrypt -> acme_certificate.

* Also renaming in BOTMETA.yml.

tonal committed:
Update pip.yml

trishnaguha committed:
nxos enable mode (#39355)

* nxos enable mode

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

* fix prompt

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

* Add authorize,auth_pass

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

* remove byte string from exec_cli_command

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

* Add on_become test

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

* removed_in_version

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

sivel committed:
Deprecate private for include_role (#39821)

Qalthos committed:
Remove provider (and transport, where applicable) from consideration when not using connection=local (#39555)

* Remove provider (and transport, where applicable) from consideration

* Add tests that misplaced transport does not fail task

piotrplazienski committed:
Don't update cpu/memory allocation in configspec if there is no change. (#39790)

Fixes #38941

mattclay committed:
Move pyvmomi install into cloud test plugin.

mattclay committed:
Mark unsupported vmware tests.

mattclay committed:
Remove unnecessary destructive aliases.

pilou- committed:
Fix unset ansible_virtualization_role (#39514)

Error was:
AttributeError("'str' object has no attribute 'startwith'",)

Closes #39138

mattclay committed:
Remove CI group from unsupported tests.

sivel committed:
Update missed private doc for import_role (#39873)

Akasurde committed:
VMware: Add more robust logic to deal with VM moref (#39462)

This fix adds more robust logic to deal with virtual machine
managed object and its comparsion.

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

mkrizek committed:
dnf: handle error when latest pkg not found (#39720)

* dnf: handle error when latest pkg not found

* Add integration test

Akasurde committed:
VMware: new module : vmware_tag_facts (#39889)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

itdependsnetworks committed:
spacing issue (#39884)

dagwieers committed:
Add meraki_team to modules/network/meraki

filicivi committed:
expect: Fix typo in examples (#39900)

smolz committed:
Update ios_system.py (#39623)

<!--- Your description here -->
Typo in the example of `domain_search`, was `domain-search`should be `domain_search`
+label: docsite_pr

kbreit committed:
Panos/doc fragment (#37696)

* Create PanOS module documentation fragment

- Module documentation fragment currently holds 3 parameters
- It most likely won't be used in every single module since there is
some variance

* Modified PanOS module for use doc_frags

- Where documentation is consistent, module documentation now uses a
documentation fragment instead of all documentation being contained in
the module.

* Formatting and syntax error fixes

Updated some formatting errors to make ansibot happy.

* Revisions for grammar

awfki committed:
Update panos_op.py (#39013)

<!--- Your description here -->
The examples were apparently pasted from panos_object and the module name wasn't updated.
+label: docsite_pr

scottmishra committed:
Update panos_security_rule.py (#39036)

<!--- Your description here -->

+label: docsite_pr

Commit:
Fix for NameError (#39665)

Issue : NameError: global name ‘cnos_devicerules’ is not defined. while running cnos modules. 
Device Rule file validates the range and type of data going into each CLI based on device type it is executed against.
This has to be backported to 2.5

mattclay committed:
Overhaul httptester support in ansible-test. (#39892)

- Works with the --remote option.
- Can be disabled with the --disable-httptester option.
- Change image with the --httptester option.
- Only load and run httptester for targets that require it.

mattclay committed:
Move lookups tests to group1.

agaffney committed:
Fix task banner with 'actionable' callback when using templates in name (#38165)

This fixes #31074.

bcoca committed:
Enabled unsafe and vault in JSON (#38759)

* allow to load json marked as unsafe or vault

 * centralized json code/decode, add vault support
 * use generics to allow for more varied inputs
 * allow inventory to dump vault w/o decrypting
 * override simplejson also
 * add entry for unsafe also
 * load vaulted and unsafe json, support unvaulting if secrets provided

agaffney committed:
Don't display host pattern warning for empty groups (fixes #35255) (#39927)

mattclay committed:
Make docs-build sanity test disabled by default.

caphrim007 committed:
Fixes for bigip_gtm_wide_ip (#39931)

Added the irules parameter. Misc corrections of invalid parameter
names and internal behaviors.

kamigerami committed:
urls: fix typo in examples section (#39777) (#39778)

ekohl committed:
Fix code snippet in playbook filters doc (#39798)

caphrim007 committed:
Adds the bigip_gtm_pool_member module (#39937)

This module can be used to manage individual pool members on a bigip's
GTM installation

dagwieers committed:
Fix module validation checks and remove old params (#39926)

This PR includes:
- Fixes to the majority of module validation issues
  (deliberate inconsistencies between docs and arg_spec)
- Removal of deprecated parameters 'method' and 'protocols'
- A few typos in the documentation

There are still some left-over validation errors, some are deliberate
(like doc strings as default to indicate ranges, etc.)

rcarrillocruz committed:
Exit admin mode properly on iosxr_config (#39594)

Fixes #38811

When using 'admin' in iosxr-config, we need to pass an end
to config terminal session but also pass exit so we exit admin
mode.

bigmstone committed:
Adding slxos_interface module (#39123)

* Adding slxos_interface module

* Fixing sanity checks

* Fixing slxos_config tests for fixture change in slxos_interface

trishnaguha committed:
remove purge from nxos_logging doc, argspec (#39947)

* remove purge from nxos_logging doc, argspec

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

* shippable

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

ikr0m committed:
Update firewalld.py (#39951)

Typo fixed

+label: docsite_pr

bigmstone committed:
Adding slxos_linkagg module (#39190)

Akasurde committed:
VMware: Custom Field in vmware_inventory (#39796)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

bigmstone committed:
Adding slxos_l2_interface module (#39309)

mkrizek committed:
git: clarify the archive option in docs (#39956)

calfonso committed:
Removing stretch items for 2.6 roadmap (#39757)

* netconf modules
  * gRPC connection plugin

caphrim007 committed:
Adds bigip gtm monitor http module (#39944)

This module can be used to manage http monitors on a GTM

rambleraptor committed:
New Module: GCP Compute Instances (#38906)

rost-d committed:
aci_l3out: Add initial ACI l3Out implementation (#37570)

* Initial implementation of ansible ACI L3Out module

* Small documentation and parameter name fixes

* fix copy-paste error

* fix sanity for version_added = 2.6

* Removed deprecated options. Added PIM protocol support

* aci_l3out: add more examples and query option

* Handle DSCP option properly

caphrim007 committed:
Adds the bigip gtm monitor tcp module (#39942)

This module can be used to manage tcp monitors on GTM

caphrim007 committed:
Adds module for F5 gtm tcp half open monitors (#39939)

This module can assist in managing TCP half open monitors in GTM

caphrim007 committed:
Adds the bigip gtm monitor https module (#39943)

This module is used to manage the HTTPS monitors on a GTM

caphrim007 committed:
Adds bigip gtm monitor firepass module (#39964)

This module can be used to manage firepass monitors on a bigip's GTM

mattclay committed:
Download files from S3 for yarn integration test.

sjmc7 committed:
Remove duplicate include in user_guide

Looks like two commits merged close to each other and an '-include'
got left along with its replacement '-include_tasks'

caphrim007 committed:
Adds the bigip gtm monitor bigip module (#39965)

This module can be used to manage bigip monitors on a bigip's GTM

caphrim007 committed:
Adds misc fixes for gtm facts module (#39966)

Codifies some parameters and fixes f5 coding conventions

jmcgill298 committed:
Update default for dscp to not allow APIC to handle it (#39970)

caphrim007 committed:
Various fixes and features for bigip modules (#39972)

* Removed forwarders parameter that did not work
* Updated coding conventions
* Added ssl_cipher_suite and ssl_protocols to bigip_device_httpd
* Added more unit tests

mattclay committed:
Mark the azure_rm_storageblob test unstable.

abadger committed:
Improve comments about file_commmon_arguments

abadger committed:
Don't pass b_path into state functions; those transform to byte strings on their own

It feels like this repeats itself because it pulls the creation of
a byte string for path into every state function.  However, it actually
cleans the API by only passing a single parameter for a thing (the path)
instead of sending it in twice.

Commit:
fixed data disk issue crash (#39716)

caphrim007 committed:
Adds bigip_device_license module (#39985)

This module can be used to license and unlicense BIGIP devices. This
does not include licensing them via BIGIQ. To do that, use BIGIQ modules

emonty committed:
Update Jesse Keating's github id (#39904)

Jesse is no longer j2sol, he's now omgjlk.

mkrizek committed:
Fix callback._dump_results to dump vaulted data v2 (#39975)

mkrizek committed:
apt_key: clarify the keyring option in docs (#39950)

tonal committed:
Correct assert test

tonal committed:
Correct test for previos task not fail

yuwzho committed:
add user-agent to Azure API calls, and support MSI/CLI cred for inventory (#39785)

* add user agent

* add msi

* support cli

* check subscription

* lint

* fix lint

* fix lint

* fix lint

Commit:
IT cases for nxos_logging (#39929)

* IT cases for nxos_logging

* review comment

felixfontein committed:
Improving documentation: make more general (less Let's Encrypt specific), and improve formatting. (#39979)

felixfontein committed:
Added botmeta for _letsencrypt symlink. (#40002)

abadger committed:
Set prev_state in the state functions

This seems a little like duplicating code since all of the called
functions need it but prev_state isn't part of argument parsing so it
doesn't belong in the toplevel main() function.

schallee committed:
Fix interfaces_file to accept allow-* (#37847)

bcoca committed:
fix comma

znantho committed:
[aws][doc] Update documentation for wait_timeout option (#39875)

Based on the documentation, 'wait_timeout' is 'Used in conjunction with instance_ids option'.  This lead me to believe that I could not use this parameter to try and solve the 'Waited too long for ELB instances to be healthy' error I was experiencing.

caphrim007 committed:
Minor fixes and additions to f5 modules (#39987)

* Fixes to docs
* Fix escaping quotes in bigip_command
* F5 coding conventions cleanup
* “warn” and “chdir” parameters added to bigip_command

lu1as committed:
"route" has to be the first option in ufw command (#31756)

For supporting deletion of ufw routes, route option has to be placed before delete option

johnroach committed:
Adding multiple project support for GCE (#39473)

* Adding multiple project support for GCP inventory.

* Adding some documentation on the .ini file

mkrizek committed:
vault: check dir existence before creating a file (#40010)

rambleraptor committed:
New Module: GCP Compute Instance Templates (#38907)

bcoca committed:
Configurable list of facts modules (#31783)

* configurable list of facts modules

 - allow for args dict for specific modules
 - add way to pass parameters
 - avoid facts poluting test
 - move to 'facts gathered' flag
 - add 'gathering' setting tests

bcoca committed:
added check mode spport to package_facts (#39801)

* added check mode spport to package_facts

fixes #39789

* also service_facts

caphrim007 committed:
Fixes and additions for f5 modules (#39986)

Small fixes in the f5 module utils. I believe the action plugins now
work consistently across types of connections

bcoca committed:
document lookup errors optoin (#39645)

* document lookup errors optoin

* changed to doc

* updated as per feedback

caphrim007 committed:
Adds the bigiq regkey license assignment module (#40025)

This module can be used to assign licenses to a bigip from a bigiq
regkey pool

caphrim007 committed:
Various bigiq fixes (#40024)

Udpdating coding conventions

caphrim007 committed:
Adds various features to bigip virtual server (#40031)

* Add type to virtual_server
* Add address_translation to virtual_server
* Add port_translation to virtual_server
* Add ip_protocol to virtual_server
* Add firewall_enforced_policy to virtual_server
* Add firewall_staged_policy to virtual_server
* Add security_log_profiles to virtual_server

caphrim007 committed:
Various bigip gtm server and vs enhancements (#40033)

* Refactor bigip_gtm_virtual_server
* Add translation_address to bigip_gtm_virtual_server
* Add translation_port to bigip_gtm_virtual_server
* Add availability_requirements to bigip_gtm_virtual_server
* Add monitors to bigip_gtm_virtual_server
* Add virtual_server_dependencies to bigip_gtm_virtual_server
* Add link to bigip_gtm_virtual_server
* Add limits to bigip_gtm_virtual_server
* Add partition to bigip_gtm_virtual_server
* Fix bigip_gtm_server to correctly create other server types

caphrim007 committed:
Various f5 modules fixes (#40037)

* Add members to bigip_gtm_pool
* Add monitors to bigip_gtm_pool
* Add availability_requirements to bigip_gtm_pool
* Refactor bigip_gtm_pool
* Normalize the product value returned by gtm facts
* Corrected various documentation
* Updated various F5 coding conventions
* Add partition to bigip_static_route
* Added more unit tests

caphrim007 committed:
Removes ignores for f5 (#40038)

Removes ignores for f5 from unit tests and fixes any issues that
are being suppressed by the ignores

pfarmer committed:
cloudstack: use paging for listVirtualMachines (#40018)

Paging wasn't implemented, so once a cs domain has over 500 (default page size) VMs, Ansible can no longer find newly created VM.

Notausstieg0309 committed:
add state parameter to patch module (#33692)

* add state parameter to patch module

* add "version_added" to state parameter docu

* fix state parameter call

* change version_added to 2.6

gundalow committed:
Network requirements to be installed by ansible-test (#40042)

Previously the test framework (DCI, Zuul) were installing the various
dependencies, this meant the list of what was required was duplicated.

Having everything defined in ansible-test makes it easier for people to
run tests locally.

Also this allows the test to work correctly on Python 2 & Python 3

resmo committed:
tests, patch: init integration tests (#40045)

* patch: init integration tests

* skip install for macos

Akasurde committed:
VMware: new module: vmware_host_powerstate (#36577)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

yuwzho committed:
location name (#40055)

SietseVisser committed:
Fix for Azure loadbalancer tags. (#39963)

* Fix for Azure loadbalancer tags.

It was possible to add tags to an Azure loadbalancer, but the tags were never set in Azure.
This patch fixes that.

* Pass shippable tests

* azure_rm_loadbalancer_facts requires rg

Getting facts of all loadbalancers via azure_rm_loadbalancer_facts requires a resource group.
This fix adds the rg as parameter to the list() call.

* Revert changes in azure_rm_loadbalancer_facts

The changes belong to another pull request.

Commit:
fix for security group description crash (#38108)

Commit:
fix nxos_ping issues (#40028)

mikewiebe committed:
Fix vxlan_vtep_vni tests (#39968)

trishnaguha committed:
ios cliconf plugin fix regex for version (#40066)

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

Akasurde committed:
VMware: Check device type explicitly (#38729)

Check datatype of device instead of comparing them directly in
vmware_guest. Also, added testcases to check this behavior.

DPVG is not supported in current version vcsim

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
icinga2_feature: fix module in check_mode (#39853)

* icinga2_feature: fix module in check_mode

* modify check_mode logic

Fixes: #39830

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
Remove redundant import from netapp module_utils (#39495)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

trishnaguha committed:
nxos module doc fragment for authorize,auth_pass (#40074)

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

robinro committed:
Pip: Make main method smaller (#38788)

robinro committed:
Allow empty list of names in pip module (#38789)

pip 10 gives exit code 1 for empty argument lists (pip < 10 gave exit 0)
see also https://github.com/pypa/pip/pull/4210

To still allow playbooks to pass when giving empty lists, don't call
pip in that case, but show a warning.

robinro committed:
Pip: Reformat tests (#38790)

Use yaml instead of old key=value format

mkrizek commented:
@tonal can you please rebase so we can proceed with merging this? Thanks!

kaechele committed:
firewalld: amend docs for python3 only hosts (#40048)

Signed-off-by: Felix Kaechele <felix@kaechele.ca>

mdellweg commented:
There is still an open question: https://github.com/ansible/ansible/pull/37973#pullrequestreview-107155369

mkrizek committed:
low_level_execute_command: add stderr_lines

dagwieers committed:
Get rid of obsolete SEQUENCETYPE-check (#40078)

And small cosmetic change to examples.

remyleone committed:
Add a User-Agent string to the API request (#38587)

mattclay commented on test/integration/targets/pip/tasks/main.yml:
```diff
@@ -1,5 +1,10 @@
+  package:
+    name: git
+  when: ansible_distribution != "MacOSX"
+  
```
This task is redundant with the one included in `pip.yml`.

suacide24 committed:
removed eth on port-channel (#40090)

Commit:
ec2_ami_copy: add tag_equality option for idempotence using tags (#40088)

* Allow idempotent use of ec2_ami_copy

When `tag_equality` is set true, use tags to determine
if AMIs in different accounts are the same, and don't
copy the AMI twice if they are the same.

Use AnsibleAWSModule and make imports more consistent
with other modules

* Update version added

* More code review changes

* Review changes - Recommended way to start EC2 connection

thedoubl3j committed:
updating win_chocolatey.py docs examples (#40101)

<!--- Your description here -->
Edit to the notepadplusplus example. 

Removed `.install` as per chocolatey documentation, the package is just notepadplusplus. 

docs found here: https://chocolatey.org/packages/notepadplusplus
+label: docsite_pr

nchepanov committed:
Add copy module support for filesystems without chmod (#40099)

* only relevant to remote_src=True
* separate file content copying from file stats copying
* raise exception on mode='preserve', ignore otherwise

Fixes: #19731
Signed-off-by: Nikita Chepanov <nchepanov@bloomberg.net>

abadger committed:
Add changelog for 40099

abadger committed:
Revert "Configurable list of facts modules (#31783)" (#40022)

This reverts commit 95655fae5c3097ec126474a1c5cbe27649b9966a.

Commit:
fix problem with documentation and param definition difference (#40067)

* fix problem with documentation and param definition difference

* removed some E324 from ignore.txt

* fixed mistake

* remove one more E324

* removed function app

* fixing append tags

* leaving append tags for later

jdufresne committed:
Update all pypi.python.org URLs to pypi.org (#38988)

For details on the new PyPI, see the blog post:

https://pythoninsider.blogspot.ca/2018/04/new-pypi-launched-legacy-pypi-shutting.html

jborean93 committed:
create module tmpdir based on remote_tmp (#39833)

* create module tmpdir based on remote_tmp

* Source remote_tmp from controller if possible

* Fixed sanity test and not use lambda

* Added expansion of env vars to the remote tmp

* Fixed sanity issues

* Added note around shell remote_tmp option

* Changed fallback tmp dir to ~/.ansible/tmp to make shell defaults

jborean93 committed:
become win: better error messages and docs update (#39936)

* become win: better error messages and docs update

* Fix syntax error and added changelog fragment

jborean93 committed:
win_template: fix issue where dest was specified as a directory (#39887)

dagwieers committed:
xml: Fix validate-modules issue (#40121)

gregsheremeta committed:
fix typo in nmcli doc (#40116)

fix typo in nmcli doc, balence -> balance

justjais committed:
Infoblox ipv6 support feature (#39668)

* Update nios.py

* Update nios.py

* Update nios.py

* nios lookup errors out when there are no results #37970  Open	

Indentation failure issue resolved

* Returning empty list instead of None

In case of no results, res will be returned as an empty list instead of None (implementing ganeshrn comment)

* infoblox ipv6 support changes

* infoblox ipv6 support changes

* for fixing pep8 errors

* moving ipaddr check to utils

* adding ipv6addr check

* increasing space to resolve pep8 error

* modified the playbook examples to valid ones

* Update nios_network.py

tonal committed:
executable joined with chdir + drop redundant code

tonal commented:
I rebase this brahch incorrectly.
Sorry.
See #40129
